### ISSUE TITLE
Better language dropdown

### ISSUE THREAD

**Issue by ericswpark** [lxndrblz/anatole#169]
Right now the language menus in the navigation bar redirect the user to the main page of the website. Would it be possible to implement the language switcher as a dropdown instead so the user can switch between the different languages of a given page? Preferably, the language switch should redirect to the same page instead of redirecting back to the main page.

Sort of like the language switcher at the bottom of 1Password's website:
<img width="362" alt="image" src="https://user-images.githubusercontent.com/13326074/112719826-4bf4a280-8f3e-11eb-8c7e-3016e7108b5f.png">

**Comment by lxndrblz** [comment]
@ericswpark Just to confirm you want this on a per-page basis rather than globally?

**Comment by ericswpark** [comment]
@lxndrblz Yup! It would be great if anatole would stay on the same page, but change to the new language. The switcher should only default to the main page if the current page does not have a translation to the selected language.

**Comment by lxndrblz** [comment]
This [dicussion](https://discourse.gohugo.io/t/how-to-translate-only-a-subset-of-the-pages/14083/5) should be a good starting point. I hope to find some time to play around with it on the weekend.

**Comment by a0m0rajab** [comment]
this [discussion](https://discourse.gohugo.io/t/change-lang-verion-on-current-page/5817/3) could help too.

**Comment by ericswpark** [comment]
@lxndrblz can we re-open this issue?

I've tried implementing the dropdown myself, but I'm not really good with web development. So far I've managed to move the language dropdown to the right of the navbar, next to the theme switcher button. The dropdown now correctly moves back and forth between translated pages instead of redirecting to the homepage, and only shows up if the current page has a translation available.

All it needs is proper CSS styling so that it shows up as an actual dropdown, but I'm not very good with CSS. Can you please take a look if you have time? It's the commit on the master branch: https://github.com/ericswpark/anatole Thanks!

**Comment by ericswpark** [comment]
<img width="1058" alt="image" src="https://user-images.githubusercontent.com/13326074/123789017-a6bfe080-d917-11eb-887f-d10dfb419d43.png">

Currently shows up like this (w/o CSS styling)

On a page with only one language version available:

<img width="1012" alt="image" src="https://user-images.githubusercontent.com/13326074/123789138-cb1bbd00-d917-11eb-93e1-f24069dce3a4.png">

**Comment by lxndrblz** [comment]
@ericswpark  Would you mind doing a PR and I will have a look at it?

**Comment by ericswpark** [comment]
Sure, will submit it right now.

**Comment by lxndrblz** [comment]
@ericswpark I will check it out and report back. The CSS shouldn't be too tricky with the new setup. It might take me a few days until I can test it  as I am really swamped with work right now.

**Comment by ericswpark** [comment]
Still active, replying so the bot won't close the issue.

**Comment by ericswpark** [comment]
@lxndrblz I've pushed another attempt at the dropdown, this time by using click instead of hover to address the previous concern of usability on mobile. So far it works OK, except for a visual bug on mobile where the dropdown triangle will show up until dismissed by clicking. I've experimented with the CSS values but I'm not sure how to fix this.

Other problems with the initial draft PR are outlined in the top post in the PR itself.

**Comment by lxndrblz** [comment]
@ericswpark Thanks. I'll check it out and try to squash that bug!

**Comment by lxndrblz** [comment]
@ericswpark I've just updated the branch with a few more commits. The dropdown functionality is now at a stage I am happy with both in terms of looks and functionality. I would love to hear your feedback!

**Comment by ericswpark** [comment]
@lxndrblz I just tried the updated branch and I really like the new implementation you made! Just two questions:

- Is the current language showing up in the dropdown intentional? In my implementation it only shows languages to switch to, to avoid confusion. Now the current language shows up as well which doesn't serve any function and takes up screen space on mobile.
- For the navbar, is there a configuration key to move the language switcher to the right of the page (next to the theme switcher button) when on a desktop? Mobile looks fine. I think for modifying the look and feel of the page placing the icons on the right-hand side is more "seamless" experience IMHO.

Can't wait for this branch to get merged. Thank you so much for all the work you've put into this!

**Comment by lxndrblz** [comment]
@ericswpark Thanks for testing it. 👍 

Regarding your remarks:

1.  Yes that was intentional, because otherwise there is an occasion, where the language switcher occurs even the page only exists in a single language. Hence, no language can be selected even though the dropdown appears. Do you have a better idea of working around that problem?
2. Well caught! I'll move it to where the theme switcher is currently located.

Thanks for your help as well!

**Comment by ericswpark** [comment]
@lxndrblz regarding point 1, my previous implementation just hid the entire language switching option if a given page did not have any translations. I checked your implementation and it was nearly there too but it was missing a `.IsTranslated` check, which was why the dropdown was showing even if the page was not translated.

Actually the `and` in this `if` statement doesn't make much sense without it:

```
{{ if and .Site.IsMultiLingual }}
```

I just changed some of the conditionals back to the initial implementation and the dropdown is correctly removed on pages with no translations. Check my branch here: https://github.com/ericswpark/anatole/tree/languagedropdown in particular this commit: https://github.com/ericswpark/anatole/commit/88c54128afd4afc7fc862d6a6ca45ad5c2b92948
